src/test/java/io/pinecone/PineconeIndexOperationClientTest.java

```diff
@@ -53,12 +58,113 @@ public void testIndexCreation() throws IOException {
         when(mockResponseFuture.toCompletableFuture()).thenReturn(mockCompletableFuture);
 
         PineconeIndexOperationClient indexDeletionService = new PineconeIndexOperationClient(clientConfig, mockClient);
-        indexDeletionService.createIndex("testEnvironment", 128, "cosine");
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest().indexName("test_name").dimension(3);
```

Comment:
What do you think about changing these builder methods to have the `with` prefix? E.g. `new CreateIndexRequest().withIndexName("my_index").withDimension(5);`

That would make them more consistent with the approach on the `PineconeClientConfig` class.

Reply:
Yes, I thought about the same before naming it the way it is but since I've started to make use of openAPI, I leaned towards their (openAPI's) naming convention. Although I agree with your point of being consistent with what is in the PineconeClientConfig already, so I'll fix the naming.

Reply:
I don't feel super strongly about it, I leave it up to you. I can see pros and cons to both ways of doing it.